rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/load/ImportLayout.java

```diff
@@ -23,76 +23,107 @@
-    private NodeSourceWindow nodeSourceWindow;
+    protected NodeSourceWindow nodeSourceWindow;
 
     private VerticalPanel importPanel;
 
     private VerticalPanel selectedImportMethodPanel;
 
     private ListBox importMethodList;
 
-    public ImportNodeSourceLayout(NodeSourceWindow nodeSourceWindow) {
+    private boolean disabled;
+
+    ImportLayout(NodeSourceWindow nodeSourceWindow, String layoutTitle, NodeSourceAction nodeSourceAction) {
         this.nodeSourceWindow = nodeSourceWindow;
-        createImportPanel();
+        this.disabled = !nodeSourceAction.isFullEditAllowed();
+        createImportPanel(layoutTitle);
         createImportMethodList();
-        addSelectImportMethodPanel();
-        addSelectedImportMethodPanel();
-    }
-
-    public void setNodeSourceWindowLabelWithError(String errorMessage, Throwable e) {
-        this.nodeSourceWindow.setNodeSourceWindowLabelWithError(errorMessage, e);
+        addImportMethodsIfEnabled();
     }
 
-    void handleNodeSourceImport(FormPanel.SubmitCompleteEvent importCompleteEvent) {
-        this.nodeSourceWindow.importNodeSourceFromJson(importCompleteEvent.getResults());
-    }
+    public abstract CatalogKind getKind();
 
-    void handleNodeSourceImport(String importedNodeSourceJsonString) {
-        this.nodeSourceWindow.importNodeSourceFromJson(importedNodeSourceJsonString);
-    }
+    public abstract void handleImport(String submitResult);
 
-    private void createImportPanel() {
-        setGroupTitle("or Import Node Source");
-        setIsGroup(true);
+    private void createImportPanel(String layoutTitle) {
         this.importPanel = new VerticalPanel();
         this.importPanel.setSpacing(10);
         this.importPanel.setHeight("70px");
-        this.importPanel.setWidth("240px");
+        this.importPanel.setWidth("272px");
+        Label importLabel = new Label(layoutTitle);
+        importLabel.setHeight("15px");
+        importLabel.setWidth("200px");
+        this.importPanel.add(importLabel);
         addMember(this.importPanel);
     }
 
     private void createImportMethodList() {
         this.importMethodList = new ListBox();
         this.importMethodList.setWidth("190px");
         this.importMethodList.addItem(GENERIC_OPTION_NAME);
-        this.importMethodList.addItem(ImportFromFilePanel.FILE_OPTION_NAME);
-        this.importMethodList.addItem(ImportFromCatalogPanel.CATALOG_OPTION_NAME);
-        this.importMethodList.addChangeHandler(this::replaceSelectedImportMethodPanelContent);
+    }
+
+    private void addImportMethodsIfEnabled() {
+        if (disabled) {
+            this.importMethodList.setEnabled(false);
+        } else {
+            this.importMethodList.addItem(ImportFromFilePanel.FILE_OPTION_NAME);
+            this.importMethodList.addItem(ImportFromCatalogPanel.CATALOG_OPTION_NAME);
+            this.importMethodList.addChangeHandler(this::replaceSelectedImportMethodPanelContent);
+        }
+        addSelectImportMethodPanel();
```

Comment:
"this" to be concise and it would help readability during github reviews.

Reply:
I find that for object attributes it really helps distinguishing from scoped variables, however for methods I cannot find a situation in which it would help, so I usually keep light on this aspect.